tools/preload.py

```diff
@@ -230,12 +233,19 @@ def fetch_webapp(app_url, directory=None):
     [appname]/metadata.json
     [appname]/update.webapp (if package_path is defined)
     [appname]/cache/ (if appcache_path is defined)
+
+    If port is > 0 that means the webapp is a hosted on
+    localhost. So we get file from filesystem or from a
+    whatever hosting site. In the latter case, icons url
+    are understood to be relative to manifest.webapp.
     """
     domain, path = split_url(app_url)
     url = urlparse(app_url)
     metadata = {'origin': domain}
     manifest_filename = 'manifest.webapp'
 
+    path_absolute_from_app_root = True if port else False
```

Comment:
`path_absolute_from_app_root` is a consequence of being hosted locally.
But as this boolean is used when calculating url for files to download, so I believe that the semantic of `path_absolute_from_app_root` is more appropriate than the one from `locally_hosted`.